src/mlpack/methods/dbscan/dbscan.hpp

```diff
@@ -80,7 +80,7 @@ class DBSCAN
    */
   template<typename MatType>
   size_t Cluster(const MatType& data,
-                 arma::mat& centroids);
+                 MatType& centroids);
```

Comment:
We should include in the function documentation (just above this declaration) that `MatType` should be a dense matrix object.  You could do `arma::Mat<ElemType>&` if you wanted (and add an additional `ElemType` template parameter too), but it's up to you.  If you do go with `arma::Mat<ElemType>&`, then it will need an additional adaptation for Bandicoot when the time comes.

Reply:
Yes, I agree it is easier to add the comment, but @rcurtin, do you think we will benefit from Bandicoot? 
Alsom would it make sense to run dbscan on GPU?

Reply:
I think there will be a benefit, but probably not if the range search uses trees. So you would have to use naive (brute force search) for the range search.